### DISCUSSION TITLE
Copy the course along with it's children

### DISCUSSION THREAD

**Discussion by somvitbhowmik** [Sunbird-Knowlg/Community#31] [General Discussions]
Hi Team, 

I am looking for the Copy API which will copy the course along with it's children. @vinukumar-vs and @Jayaprakash8887 any idea or suggestion on this. 

Regards,
Somvit

**Comment by Jayaprakash8887** [comment]
Have you tried http://docs.sunbird.org/latest/apis/contentapi/index.html#operation/Copy%20Content ?

Pls try {{host}}/api/content/v1/copy/{{contentId}}

**Reply by somvitbhowmik** [reply]
Yes @Jayaprakash8887 have tried. It is creating a copy of the course or resource. But I need to know is there any api in sunbird which will create a copy of a course along with resources which is tagged in a course.

**Reply by Jayaprakash8887** [reply]
copy api copies collection along with resources linked to it.

**Reply by Jayaprakash8887** [reply]
are u referring to a case where a course has a course linked to it?

**Reply by pkranga** [reply]
@Jayaprakash8887 The need is that the whole hierarchy has to be created as a fresh content. Currently the course is copied to a new do_id  and the courseunit are used as existing do_ids. The resources are also the same do_id. We need to have a copy of all the do_id's not just course

**Reply by Jayaprakash8887** [reply]
@pkranga copy API recreates Hierarchy including units. Please make sure to not send '?type=shallow' in the URL.

**Reply by pallakartheekreddy** [reply]
@somvitbhowmik As we haven't received a response from you, we will proceed to close this thread. If you have any further questions or require assistance in the future, please don't hesitate to start a new thread. Thank you.
